### PR TITLE
Add filtering to references API

### PR DESCRIPTION
**Pull request by michaelb990**
Signed-off-by: Michael Brown <brownxmi@amazon.com>

### CODE REVIEW

**Review by sajayantony — APPROVED** [review]

File: spec.md [hunk 1]
```diff
@@ -522,8 +533,40 @@ Upon success, the response MUST be a json body in the following format (an index
...
+
+The registry SHOULD support filtering on `artifactType`. 
+If filtering is requested and applied, the response MUST include an annotation (`org.opencontainers.references.filtersApplied`) denoting that an `artifactType` filter was applied.
+
+Example request with filtering:
+```
+GET /v2/<name>/referrers/<ref>?artifactType=application/vnd.example.icecream.v1
+```
+Response:
+```json
```
Comment (sudo-bmitch) [inline spec.md]: switch to jsonc for syntax highlighting of comments
```suggestion
```jsonc
```

**Review by sudo-bmitch — CHANGES_REQUESTED** [review]
Rebase to pickup DCO fixes, we'll also need to add the endpoint API changes, but that's can be a separate commit.

**Review by jdolitsky — APPROVED** [review]

**Comment by jdolitsky** [comment]
> we'll also need to add the endpoint API changes

@sudo-bmitch - sorry confused.. are you referring to [endpoints table](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#endpoints)? separating into a 12a/12b ?

**Comment by sudo-bmitch** [comment]
> > we'll also need to add the endpoint API changes
> 
> @sudo-bmitch - sorry confused.. are you referring to [endpoints table](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#endpoints)? separating into a 12a/12b ?

@jdolitsky Yup. (Much to your likely disappointment.) One line for the API without a filter parameter, and one with. Once we get this rebased and merged, I'm happy to tackle that since I remember all the things we broke doing it before.

**Comment by michaelb990** [comment]
I think I've made the updates requested. Anything else? @sudo-bmitch @jdolitsky

**Review by jdolitsky — APPROVED** [review]
looks fantastic

**Review by sudo-bmitch — APPROVED** [review]
Looks good, I'm going to pull this one in and then rebase my changes.

**Pull request merged**